### ISSUE TITLE
OSX - Google Office Docs extension loads as text - OK on Win 10 (sdk)

### ISSUE THREAD

**Issue by Blatman** [nwjs/nw.js#7010]
MacOS 10.13.6 High Sierra using NW.js 0.36.5-sdk and 0.37.0-sdk

OSX version will load extension but only as text - will work OK if using the sdk version on win10
Tried .xls and .xlsx - both fail.

MacOS Fail -
![osxfail](https://user-images.githubusercontent.com/11089335/54084263-8c1c6480-4326-11e9-8302-e8a64311572f.jpg)

Win OK -
![winok](https://user-images.githubusercontent.com/11089335/54084265-92aadc00-4326-11e9-91d9-cf009b5ded33.jpg)

[TestExtensions2.zip](https://github.com/nwjs/nw.js/files/2949372/TestExtensions2.zip)

Extension can be used from example in #7008 

Also tried nw2 switch and no difference

Cheers

**Comment by ffanny** [comment]
Do you use an absolute path in OSX?

I tried with nwjs-sdk-linux & win.
When using a relative path, win & Linux both load text rather than the office file.
When using an absolute path, nwjs-sdk-win loads the file correctly.

**Comment by rogerwang** [comment]
Thanks for the hint from @ffanny . Yes the chrome-extension:// protocol won't return MIME types provided by the OS. See https://github.com/nwjs/chromium.src/blob/nw37-log/extensions/browser/extension_protocols.cc#L372
However, we could consider workaround in that code.

**Comment by Blatman** [comment]
Yup OK - tried the absolute and it works OK.
eg:
src="file:///Users/USER/Desktop/TestExtensions/test.xls"  works OK
src="./test.xls" or src="test.xls" fails

Thanks

**Comment by rogerwang** [comment]
This is fixed in git and will be available in the next [nightly build](http://dl.nwjs.io/live-build/).

**Comment by Blatman** [comment]
Tried latest -
Index of /live-build/nw37/03-12-2019/81ceb52bf/v0.37.0/
doesn't work with relative (src=".test.xls") - not sure if intention of git fix was to allow this

Cheers

**Comment by rogerwang** [comment]
Could you find out the mime mapping in your system by `file -I test.xls`?

**Comment by Blatman** [comment]
@rogerwang - result was -

test.xls: application/vnd.ms-excel; charset=binary

test.xls generated from Office for Mac 2011 saved as 97-2004 workbook(.xls)
Tried different file from LibreOffice (saved as 97-2003 (.xls)) and was the same.

**Comment by Blatman** [comment]
Just had a fossick about and found some clues -
This site was discussing similar issue and felt may be chromium issue -
https://github.com/pqina/filepond-plugin-file-validate-type/issues/4

Chromium reference in above post -
https://bugs.chromium.org/p/chromium/issues/detail?id=155455&q=xls&colspec=ID%20Pri%20M%20Stars%20ReleaseBlock%20Component%20Status%20Owner%20Summary%20OS%20Modified

I tried a .doc file as a relative and it worked so maybe the issue is only with .xls on chromium
src="./test.doc"          test.doc: application/msword; charset=binary

Cheers

**Comment by ffanny** [comment]
I tried with latest nightly build nwjs-sdk-v0.37.0-win ([Index of /live-build/nw37/03-12-2019/81ceb52bf/v0.37.0/](https://dl.nwjs.io/live-build/nw37/03-12-2019/81ceb52bf/v0.37.0/)) and `src="./test.xls" src="./test.csv" src="./test.pptx"`  all work well.

**Comment by Blatman** [comment]
@ffanny 
have you got Excel installed on your machine? I noticed in the chromium bug report the following -
*****
After some investigation, I have not solved this issue but I found some relevant info:

On Macs without Microsoft Excel installed, the mime type works fine.
However, on Macs with Excel installed or previously installed, the mime type shows as null!
This explains why larh...@gmail.com wasn't able to reproduce on a fresh install of macOS

I ran this command:

file --mime Workbook1.xls 

on an xls document on our macs. The strange thing is that all computers showed the same thing:

Workbook1.xls: application/vnd.ms-excel; charset=binary
*****

Excel is installed on all my machines (note: have cleared profile of test app prior to testing)
